.ci/openshift-ci/cluster/deployments/relabel_selinux.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: relabel-selinux-daemonset
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: restorecon
+  template:
+    metadata:
+      labels:
+        app: restorecon
+    spec:
+      serviceAccountName: kata-deploy-sa
+      hostPID: true
+      containers:
+        - name: relabel-selinux-container
+          image: alpine
+          securityContext:
+            privileged: true
+          command: ["/bin/sh", "-c", "nsenter --target 1 --mount bash -xc \"for ENTRY in '/(.*/)?opt/kata/share/kata-.*(/.*)?(/.*)?' '/(.*/)?opt/kata/share/ovmf(/.*)?' '/(.*/)?opt/kata/share/tdvf(/.*)?' '/(.*/)?opt/kata/libexec(/.*)?'; do semanage fcontext -a -t qemu_exec_t \\\"\\$ENTRY\\\"; done; restorecon -v -R /opt/kata\"; echo NSENTER_FINISHED_WITH: $?; sleep infinity"]
```

Comment:
I don't quite understand the motivation behind each regexp. Also on a typical RHEL or RHCOS setup, only the QEMU binary gets the `qemu_exec_t` label. Other files have different labels, e.g. `virtiofsd` has `bin_t`.

Can you clarify @ldoktor ?

Reply:
Hello Greg, I used `qemu_exec_t` as that one is defined in `container-selinux` and contains the `map` and `execute` privileges, which is what we need for binaries and bios files. The `bin_t` would do as well but might be too generic.

As for the `/(.*/)?opt` part it's because selinux policy uses `realpath` and `/opt` is mounted from `/var/opt` on azure so we need to allow this extra flexibility.

There might be few shortcuts as it should only serve the CI pipeline. Once we hammer-out the details the proper implementation should become part of https://github.com/kata-containers/kata-containers/pull/8417 Anyway I'm listening so if you know how to do things properly, please let me know (I'm just a selinux user, not a policy writer)

Reply:
In the real OCP cluster we rely on system-wide installed tools (qemu) while in the CI pipeline we use the latest built versions that are deployed via `kata-deploy`. The problem is that currently `kata-deploy` does not cover the `selinux` labels at all which results in selinux rejects.

For the rules I just collected the required privileges and found the "closest matching label" and, for consistency, enabled it on all executable/binary files delivered by the `kata-deploy` script, although I only tested qemu. This was sufficient for e2e pipeline to succeed while keeping selinux enabled in the CI, which is a step forward.

Obviously after the `kata-deploy` script is improved this CI-pipeline "hack" should be removed but for now it might serve as an incubator to see what all is needed in `kata-deploy`.

Reply:
Thanks for the clarifications @ldoktor.